### PR TITLE
New court report template for montgomery

### LINKED ISSUE #2265: Use new Montgomery court report templates instead of old one
Use new Montgomery court report templates instead of old one

The current Montgomery court report template is: https://github.com/rubyforgood/casa/blob/main/app/documents/templates/montgomery_report_template.docx

New court report templates are at: 
1. over 14 (transition aged youth) https://drive.google.com/file/d/1B89zt95qAP5pg-qHQTi05Ec_1sPjYG66/view?usp=sharing
2. under 14 (non transition aged youth) https://drive.google.com/file/d/1dJVlQ4wIoRO79sE1fwsDFYtCa-Zoy2oP/view?usp=sharing

Please merge these into the same template

Steps:
1. Edit the new files to have the correct `«=casa_case.court_date»` variables (it might be easier to start with the old file and copy over the new info into the file) Remember to show different info for transition aged youth vs not- you can use case statements or move some of the text into the codebase if that helps- keep in mind that we want to avoid writing custom code for each CASA as much as possible because it's not scalable)
1. On your local, generate a court report of each type and check that all the variables are set correctly so the dates and names display etc
1. Write even MORE tests in addition to the existing tests, since you're in here anyway. Test for edge cases like missing data and extra-long variable values. 
1. Ask in slack when you have questions! https://rubyforgood.slack.com/ `#casa` channel
1. Update https://github.com/rubyforgood/casa/wiki/Making-a-new-Court-Report-Format with your learnings on how to do this :)

see https://github.com/rubyforgood/casa/wiki/How-to-edit-docx-templates

### PR DESCRIPTION
**Pull request by FireLemons**
### What github issue is this PR for, if any?
Resolves #2265

Waiting on more input from stakeholders

### What changed, and why?
New template contains old template's autofilled information
Warning text for if no case contacts or mandates

### How is this tested? (please write tests!) 💖💪
Not testing until design is finalized

### Screenshots please :)
Generated court report without case contacts or mandates
image

### CODE REVIEW

**Comment by compwron** [comment]
Very cool!!!

File: app/models/case_court_report.rb [hunk 1]
```diff
@@ -21,11 +21,14 @@ def generate_to_string
   private
```
Comment (compwron) [inline app/models/case_court_report.rb:21]: test failure looks like a flake
> 1) Seeds test development DB successfully populates all necessary tables
     Failure/Error: ActiveRecord::Tasks::DatabaseTasks.load_seed

     ActiveRecord::RecordInvalid:
       Validation failed: Volunteer and case must belong to the same organization

File: app/models/case_court_report.rb [hunk 2]
```diff
@@ -21,11 +21,14 @@ def generate_to_string
...
 
   def prepare_context(is_default_template)
+    case_past_court_dates = @casa_case.past_court_dates
+
     {
       created_date: I18n.l(Date.today, format: :full, default: nil),
       casa_case: prepare_case_details,
       case_contacts: prepare_case_contacts,
       case_mandates: prepare_case_mandates,
+      latest_hearing_date: case_past_court_dates.any? ? I18n.l(case_past_court_dates.where(date: case_past_court_dates.select("MAX(date)"))[0].date, format: :full, default: nil) : "___<LATEST HEARING DATE>____",
```
Comment (compwron) [inline app/models/case_court_report.rb:31]: It would be nice if this logic were in a method

File: app/models/case_court_report.rb [hunk 3]
```diff
@@ -44,7 +47,7 @@ def prepare_case_contacts
     contact_dates_as_hash = aggregate_contact_dates(interviewees)
     contact_dates_as_hash.map do |type, dates|
       {
-        name: "Firstname Lastname",
+        name: "Names of persons involved starting with the child's name",
```
Comment (compwron) [inline app/models/case_court_report.rb:50]: I would like to see a comma between the words `involved` and `starting`

File: lib/tasks/deployment/20210811052058_update_montgomery_court_report_template.rake [hunk 4]
```diff
@@ -0,0 +1,11 @@
+namespace :after_party do
+  desc "Deployment task: update_montgomery_court_report_template"
+  task task_name: :environment do
+    CasaOrg.where(name: "Voices for Children Montgomery").map do |casa_org|
```
Comment (compwron) [inline lib/tasks/deployment/20210811052058_update_montgomery_court_report_template.rake:4]: nice usage of afterparty

File: spec/models/case_court_report_spec.rb [hunk 5]
```diff
@@ -67,6 +67,18 @@
...
       end
+
+      context "when the case has multiple past court dates" do
+        before do
+          casa_case_with_contacts.past_court_dates << create(:past_court_date, date: 9.months.ago)
+          casa_case_with_contacts.past_court_dates << create(:past_court_date, date: 3.months.ago)
+          casa_case_with_contacts.past_court_dates << create(:past_court_date, date: 15.months.ago)
+        end
+
+        it "sets latest_hearing_date as the latest past court date" do
```
Comment (compwron) [inline spec/models/case_court_report_spec.rb:78]: yay test!!

File: spec/requests/case_court_reports_spec.rb [hunk 6]
```diff
@@ -116,7 +116,6 @@
         it "contains a link ending with .DOCX extension" do
           body_hash = JSON.parse(response.body)
 
-          expect(body_hash["link"].length).to be > 0
```
Comment (compwron) [inline spec/requests/case_court_reports_spec.rb:119]: Why did this go away?

File: spec/requests/case_court_reports_spec.rb [hunk 7]
```diff
@@ -125,6 +124,17 @@
 
           expect(get_docx_subfile_contents(response.body, "word/header3.xml")).to include("YOUR CASA ORG’S NUMBER")
         end
+
+        # TODO admin and supervisor tests instruction in wrong order
```
Comment (compwron) [inline spec/requests/case_court_reports_spec.rb:128]: yes tests please

File: spec/requests/case_court_reports_spec.rb [hunk 8]
```diff
@@ -174,18 +184,7 @@
       it "uses the custom template" do
         get JSON.parse(response.body)["link"]
 
-        expect(get_docx_subfile_contents(response.body, "word/header2.xml")).to include("Voices for Children Montgomery")
```
Comment (compwron) [inline spec/requests/case_court_reports_spec.rb:177]: We should keep this test in addition to the new test

**Review by compwron — APPROVED** [review]
Exciting!
I don't see the template itself in this PR, remember to commit it?

**Comment by compwron** [comment]
To test this adequately in qa and staging, let's make sure that that afterparty task runs - by renaming one of the orgs in each environment to have this name

Also after deploy we should check that this works in prod

**Comment by FireLemons** [comment]
> Exciting!
> I don't see the template itself in this PR, remember to commit it?

The new template is `app/documents/templates/montgomery_report_template.docx `

File: spec/requests/case_court_reports_spec.rb (hunk 8 above)
Comment (FireLemons) [reply spec/requests/case_court_reports_spec.rb:177]: The new template doesn't have "Voices for Children Montgomery" and when you extract the xml from the docx there's no `word/header2.xml`

File: spec/requests/case_court_reports_spec.rb (hunk 7 above)
Comment (FireLemons) [reply spec/requests/case_court_reports_spec.rb:128]: out of the scope of this issue and the test is wrong because it generates the docx download link as a volunteer

File: spec/requests/case_court_reports_spec.rb (hunk 8 above)
Comment (FireLemons) [reply spec/requests/case_court_reports_spec.rb:177]: the `as a supervisor` test is moved up and commented out

File: spec/requests/case_court_reports_spec.rb (hunk 7 above)
Comment (compwron) [reply spec/requests/case_court_reports_spec.rb:128]: interesting... can you make a chore about it plz

File: app/models/case_court_report.rb (hunk 1 above)
Comment (FireLemons) [reply app/models/case_court_report.rb:21]: I've seen that in least least one other pr
